ci/deploy-gh-pages.sh

```diff
@@ -28,12 +24,11 @@ else
 fi
 
 echob "Build doxygen documentation."
-if ! ( cd $script_path && doxygen "Doxyfile" ); then
-    echo "- Doxygen generation failed."
-    exit 1
-fi
+test cd $doc_path
```

Comment:
cd is not necessary, you can call doxygen with the correct path as argument.

Reply:
This is a non-functional change to this script, it is the same logic as previously.

Reply:
if we don't cd, then the output is created in another folder, and the paths/logic in the remainder of this script must change. This script needs an update generally, it can be streamlined to not re-clone.